docs/architecture/adr-056-proving-amnesia-attacks.md

```diff
@@ -23,10 +24,98 @@ The currently only known form of global evidence stems from [flip flopping](http
+
+## Detailed Design
+
+The first part of this short term solution is the `PotentialAmnesiaEvidence` data structure. This has the following properties that differ from its prior implementation.
+
+- It bundles all the malicious validators together instead of splitting them into individual evidence
+- It can be submitted on the chain
+- It does, by itself, not indicate which validators in the set misbehaved and should be slashed. It is merely a prompt for manual investigation.
+
+Before going any further, it is also important to mention that `PotentialAmnesiaEvidence` should only be formed from a valid attack on the light client and measures should be put in place to ensure that no node can easily forge the evidence as a means of spamming the network. 
+
+The data structure for `PotentialAmensiaEvidence` is as follows:
+
+```golang
+type PotentialAmnesiaEvidence struct {
+	*SignedHeader
+}
+```
+
+It conforms to the `Evidence` interface where `Time()` is the time of the heaer (note that this can be forged but must be within trusting period) and `Address()` returns the bytes of all validators in the commit
+
+`ValidateBasic()` inherit from the validate basic of the signed header
+
+### Verification of PotentialAmensiaEvidence
+
+`PotentialAmensiaEvidence` will be saved in the evidence pool and committed on chain if the following conditions are met.
+
+- The `SignedHeader` is valid -> `ValidateBasic()`
+
+- The `ValidatorsHash` of the header must be the same as the header that the node has committed. (else this is a lunatic attack)
+
+- The signatures of the commit must all be valid (for that header) `VerifyCommit()` using the validator set the node has for that height.
+
+- The header hash must be different to the hash of the header that the node has.
+
+- The `Commit` of the `SignedHeader` must be for a different round than the `Commit` of the header that the node has committed
+
+- The evidence must not have expired.
+
+
+The second part of the short term solution is saving `VoteSet`'s to the evidence pool. In order to avoid overflowing the validators memory, only the relevant information will be taken from the voteSets and formed into
+this specific struct
+
+```golang
+type VotesRecord struct {
+	votes []*Vote
+}
+```
+
+These votes should all be precommit votes of the same height and round and will come from the consensus reactor. `VotesRecord` will only be created in heights where there are more than 1 round and will be sent from consensus. 
+`VotesRecord` will also follow the same pruning algorithm as the rest of the evidence, being removed after expiring. 
+
+Implementing this is in the short term should be sufficient to detecting amnesia attacks using manual intervention which through off-chain conesnsus can lead to punishment and thus in itself act to disincentivise misbehaviour.
```

Comment:
will Tendermint instance stop upon seeing & committing such evidence?

Reply:
This is a good question and I don't think it was one that was resolved. I think mainly the use of keeping amnesia evidence is to punish the light client version of that attack (back to the past). Because of this I don't think that nodes should stop just because they see the evidence (also how much extra damage can be caused if nodes do continue to operate if the fork has already occured)

Reply:
My understanding is that we consider light client forks as something sand-boxed so it does not propagate into full node. So it only lives inside evidence reactor so it should not stop the full node.

Reply:
But amnesia can also be a full fork